GoogleSignIn/Sources/GIDAuthorizationFlowProcessor/API/GIDAuthorizationFlowProcessor.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class GIDSignInInternalOptions;
+@class OIDAuthorizationResponse;
+
+/// The protocol to control the authorization flow.
+@protocol GIDAuthorizationFlowProcessor <NSObject>
+
+/// The state of the authorization flow.
+@property(nonatomic, readonly, getter=isStarted) BOOL start;
+
+/// Starts the authorization flow.
+/// This method sends authorization request to AppAuth OIDAuthorizationService and gets back the response or an error.
+///
+/// @param options The @GIDSignInInternalOptions object to provide serverClientID, hostedDomain,
+///     clientID, scopes, loginHint and extraParams.
+/// @param emmSupport The EMM support info string.
+/// @param completion The block that is called on completion asynchronously.
+- (void)startWithOptions:(GIDSignInInternalOptions *)options
+              emmSupport:(nullable NSString *)emmSupport
+              completion:(void (^)(OIDAuthorizationResponse *_Nullable authorizationResponse,
+                                   NSError *_Nullable error))completion;
+
+/// Handles the custom scheme URL opened by SFSafariViewController to cleanup UI on iOS 10.
```

Comment:
Let's not forget the Device Policy app and the EMM case.  Also, this path is returning control to the 3P app, not sure what "cleanup UI" is referring to.

Reply:
In GIDSignIn code the Device Policy app is handled by this this method. This is only for custom custom URL.

Reply:
Removed `cleanup UI`.